### PR TITLE
fix(breaking_change): default to headless and 1280x720 for cypress run in all browsers

### LINKED ISSUE #16832: All browsers when run via `cypress run` should run `headless` by default.
### What would you like?

All browsers when run via `cypress run` should run `headless` by default.

Currently, whether a browser is run `headless` or `headed` by default is not consistent. It depends on the browser. We originally made this decision in order to not introduce a breaking change when Chrome headless was delivered. https://github.com/cypress-io/cypress/pull/5676 We intended to change this behavior in the next breaking change, but clearly didn't prioritize this work. 

- `cypress run --browser electron` - defaults to headless
- `cypress run --browser chrome` - *defaults to headed*
- `cypress run --browser firefox` - *defaults to headed*
- `cypress run --browser edge` - *defaults to headed*

### Why is this needed?

The larger reason this is necessary (besides maintaining consistency in our app) is because the browser size when Chrome is opened in headed mode is unpredictable. 

This causes a lot of unexpected video/screenshot sizes when someone wants to run their browser in chrome, not considering the (undocumented) differences between headed and headless screen sizes. 

People might additionally believe that passing browser flags for the size will fix the Chrome headed browser to a specific size when in reality it does absolutely nothing.

Chrome video sizes (browser size) during headed:

- Screensize: 2880x1800 at 2dpi = **2560x1050**
- Screensize: 1024x768 at 1dpt = **896x528**
- Xvfb size: 1280x1024 = **1050x838**
- Xvfb size: 1024x768 = **1004x582**

### LINKED ISSUE #16853: Make default screen size for headless runs to be 1280x720
### What would you like?

The default screen size for headless runs in Cypress should be 1280x720.

### Why is this needed?

In an attempt to improve the resolution of screenshots and videos, we updated the resolution of videos to be 1920x1080 in Cypress 7.0: https://github.com/cypress-io/cypress/pull/15686

After running many tests in different OS's, containers, screen sizes, hiDPI, etc, **we've concluded that this previous change did not improve the resolution of videos. In fact, it made the videos and screenshots less readable as the content is now tiny.**

Making the breaking change to set the size back to 1280x720 is part of a larger part of stories to improve default video and screenshot resolution. See https://github.com/cypress-io/cypress/issues/16832

#### Before 7.0 (1280x720)

https://user-images.githubusercontent.com/1271364/121219682-9309e880-c849-11eb-900e-54ad18ec567e.mp4

#### After 7.0 (1920x720)

https://user-images.githubusercontent.com/1271364/121219769-a7e67c00-c849-11eb-8ce1-35b35162337e.mp4

### PR DESCRIPTION
**Pull request by kuceb**
- Closes #16832 
- Closes #16853 
### User facing changelog

Breaking Change: default to headless mode in all browsers during `cypress run`. Previously Chrome and Firefox browsers defaulted to headed, whereas electron defaulted to headless. 
Breaking Change: headless browsers now default to 1280x720 instead of 1920x1080
### PR Tasks
- [x] cy.type tests for contenteditable elements failing in headless firefox, revealed now that our own tests are running in headless
  - [x]  fix to ensure firefox headless gains window focus and keeps it

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
I changes the base branch to 8.0

**Comment by Bkucera** [comment]
unit tests are failing in release branch

**Comment by Bkucera** [comment]
should be good to go

File: packages/driver/src/cypress.js [hunk 1]
```diff
@@ -230,6 +230,17 @@ class $Cypress {
         }
       }))
     })
+    .then(() => {
+      if (this.browser.isHeadless && this.isBrowser({ family: 'firefox' })) {
```
Comment (chrisbreiding) [inline packages/driver/src/cypress.js:234]: Could you add a code comment here about why this is necessary?

**Review by chrisbreiding — APPROVED** [review]
LGTM 👍

**Pull request merged**